event/src/main/scala/com/improving/app/event/domain/Event.scala

```diff
@@ -139,34 +132,57 @@ object Event extends StrictLogging {
                   x match {
                     case ScheduledEventState(info, meta) =>
                       command.request match {
-                        case startEventCommand: StartEvent => startEvent(info, meta, startEventCommand)
-                        case delayEventCommand: DelayEvent => delayEvent(Right(info), meta, delayEventCommand)
+                        case startEventCommand: StartEvent =>
+                          if (Instant.now().getEpochSecond > info.getExpectedStart.seconds)
+                            startEvent(info, meta, startEventCommand)
+                          else
+                            Left(
+                              StateError(
+                                s"${command.request.productPrefix} command cannot be used before an Event has started"
+                              )
+                            )
                         case rescheduleEventCommand: RescheduleEvent =>
                           rescheduleEvent(info, meta, rescheduleEventCommand)
+                        case delayEventCommand: DelayEvent   => delayEvent(Right(info), meta, delayEventCommand)
+                        case cancelEventCommand: CancelEvent => cancelEvent(Right(info), meta, cancelEventCommand)
                         case editEventInfoCommand: EditEventInfo =>
                           editEventInfo(Right(info), meta, editEventInfoCommand)
                         case _ =>
                           Left(
-                            StateError(s"${command.request.productPrefix} command cannot be used on an active Event")
+                            StateError(s"${command.request.productPrefix} command cannot be used on a scheduled Event")
                           )
                       }
                     case InProgressEventState(info, meta) =>
                       command.request match {
                         case delayEventCommand: DelayEvent =>
                           delayEvent(Right(info), meta, delayEventCommand)
                         case endEventCommand: EndEvent =>
-                          endEvent(meta, endEventCommand)
+                          if (Instant.now().getEpochSecond > info.getExpectedEnd.seconds)
```

Comment:
maybe should be `>=` ?

Reply:
And similarly on the end, ending early is a thing that happens sometimes and it might be nice to be able to do that without a big rescheduling workflow